module/PowerNSX.psm1

```diff
@@ -22920,58 +22920,54 @@ function Remove-NsxIpSetMember  {
     process {
 
         $_ipset = $ipset.clonenode($true)
-        if ( -not (invoke-xpathquery -QueryMethod SelectSingleNode -Node $_ipset -query "child::value")) {
-            throw "IPSet $($ipset.name) has no members."
-        }
-
-        if ( $_ipset.value -eq "" ) {
-            throw "IPSet $($ipset.name) has no members."
-        }
-
-        [system.collections.arraylist]$ValCollection = $_ipset.value -split ","
         $modified = $false
-        foreach ( $value in $IPAddress ) {
-            # An IPSET allows the users to enter a host as either 1.1.1.1 or
-            # 1.1.1.1/32. So if the users specifies that they want to remove
-            # 1.1.1.1 we need to look for both 1.1.1.1 AND 1.1.1.1/32 to remove.
-            if ( ValidateIPHost $value ) {
-                if ( $value -as [ipaddress] ) {
-                    if ( ( -not ( $valcollection -contains $value ) ) -and ( -not ( $valcollection -contains "$($value)/32" ) ) ) {
-                        write-warning "$Value not a member of IPSet $($ipset.name)"
+        if ( ( $_ipset.value -eq "" ) -or ( -not (invoke-xpathquery -QueryMethod SelectSingleNode -Node $_ipset -query "child::value")) ) {
```

Comment:
You should test that the value node exists separately (and before) testing its value.